### ISSUE TITLE
LayerNorm param_shape inconsistent with tensorflow/paper

### ISSUE THREAD

**Issue by jackd** [deepmind/dm-haiku#186]
[`LayerNorm`'s uses the final dimension of inputs to define `param_shape`](https://github.com/deepmind/dm-haiku/blob/main/haiku/_src/layer_norm.py#L126). This is inconsistent with both [tensorflow's implementation](https://github.com/tensorflow/tensorflow/blob/v2.5.0/tensorflow/python/keras/layers/normalization.py#L1186), and my understanding of the original paper. haiku's implementation looks closer to [sonnet's](https://github.com/deepmind/sonnet/blob/v2/sonnet/src/axis_norm.py#L193).

If this different is intentional, clear documentation would be appreciated.

**Comment by trevorcai** [comment]
Confirming this is (silently!) wrong. @tomhennigan probably worth a fix.

**Comment by tomhennigan** [comment]
Thanks for bumping this @trevorcai and for reporting this @jackd. I am not sure why I missed this when originally reported.

This is indeed something we have carried into Haiku from Sonnet 1 (via Sonnet 2). The Sonnet 1 impl from 2018 is here https://github.com/deepmind/sonnet/blob/cd5b5fa48e15e4d020f744968f5209949ebe750f/sonnet/python/modules/layer_norm.py#L142-L144

I'll look into what we should do here and report back.
